### ISSUE TITLE
error: src refspec master does not match any.

### ISSUE THREAD

**Issue by maormagori** [Stremio/stremio-beamup-cli#1]
Hello, I was trying to deploy an addon using the CLI but the got this error:

```
> git push beamup master
error: src refspec master does not match any.
error: failed to push some refs to 'dokku@<HOST>/<PROJECT'S NAME>'
```

The workaround I found is to push and indicate the refspec like so:
```
git push dokku@<HOST>/<PROJECT'S NAME> HEAD:master
```

I don't believe it's a problem with the cli but with git. Just wanted to help anyone else who encounters this problem.

**Comment by jaruba** [comment]
Thank you for reporting it, that's an interesting find, what OS are you running, and what git version are you using? (you can find the git version by using `git --version`)

**Comment by maormagori** [comment]
Beamup version: 1.1.4
Git version: 2.19.0.windows.1
Windows version: Windows 10 x64

**Comment by jaruba** [comment]
Btw, the error you got "failed to push some refs to ..." can also happen (if i'm not mistaken) if you are trying to redeploy (update) a project and did not make any new commits since the last time you deployed. this happens because of the underlying use of `git`, the server can only accept new deployments if the commit history changed.

I'm planning to check against this situation in the future, and make the cli ask if you want to create a new commit if the commit history did not change.

Are you sure this was not the case for your attempt to deploy also?

**Comment by maormagori** [comment]
>  can also happen (if i'm not mistaken) if you are trying to redeploy (update) a project and did not make any new commits since the last time you deployed.

This happened on first deployment so I don't this is the case. But just to be sure I made a new commit and redeployed and got the same result.

**Comment by jaruba** [comment]
What are the chances that this is caused by this:
https://www.zdnet.com/article/github-to-replace-master-with-main-starting-next-month/

GitHub changed the default branch name from `master` to `main`, maybe `git push beamup master` fails because your primary branch is actually called `main` now.

**Comment by jaruba** [comment]
Try using `git push beamup HEAD:master`

**Comment by maormagori** [comment]
It works!
I knew it was something silly that caused it... 😅

Maybe change master to main? in:
https://github.com/Stremio/stremio-beamup-cli/blob/ac346e486e736bd65b2f6b423aa223dabd219d7a/lib/deploy.js#L22

Or even better, run `git branch --show-current` and read the output?

**Comment by jaruba** [comment]
Or we could just use `git push beamup HEAD:master`, this command should work regardless of if the primary branch is called `master` or `main`

**Comment by jaruba** [comment]
@maormagori I released `beamup-cli` v1.1.5 with the change, please test it when you can

**Comment by maormagori** [comment]
Flawlessly! 👍
